bitnami/nginx/templates/http-block-configmap.yaml

```diff
@@ -0,0 +1,19 @@
+{{- /*
+Copyright VMware, Inc.
+SPDX-License-Identifier: APACHE-2.0
+*/}}
+
+{{- if and .Values.httpBlock (not .Values.existingHttpBlockConfigmap) }}
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ template "common.names.fullname" . }}-http-block
+  namespace: {{ include "common.names.namespace" . | quote }}
+  labels: {{- include "common.labels.standard" ( dict "customLabels" .Values.commonLabels "context" $ ) | nindent 4 }}
+  {{- if .Values.commonAnnotations }}
+  annotations: {{- include "common.tplvalues.render" ( dict "value" .Values.commonAnnotations "context" $ ) | nindent 4 }}
+  {{- end }}
+data:
+  nginx.conf: |-
```

Comment:
Would it make sense to rename the values, helper methods, examples, files... from the "http block" concept to something else? I'm saying this because all this implementation is actually a way to overwrite the main NGINX configuration file which exceeds the limits of the http context.

What if we rename it to `configuration` / `existingConfigurationConfigmap`, etc?

Let us know if that would make sense for you.


By the way, I guess you are aware of the existing `extraVolumes` / `extraVolumeMount` options to provide a custom NGINX configuration file with the existing values, right? It is fine to have a new one specific for the configuration though, just letting you know of this other option.